File: android/src/main/java/org/maplibre/reactnative/components/annotation/MLRNCalloutManager.java
package org.maplibre.reactnative.components.annotation;

import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;

/**
 * Created by nickitaliano on 10/11/17.
 */

public class MLRNCalloutManager extends ViewGroupManager<MLRNCallout> {
    public static final String REACT_CLASS = "MLRNCallout";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNCallout createViewInstance(ThemedReactContext reactContext) {
        return new MLRNCallout(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/annotations/MLRNCalloutManager.kt
package org.maplibre.reactnative.components.annotations;

import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import org.maplibre.reactnative.components.camera.MLRNCameraManager

class MLRNCalloutManager : ViewGroupManager<MLRNCallout>() {
    companion object {
        const val REACT_CLASS: String = "MLRNCallout"
    }

    override fun getName(): String = REACT_CLASS

    protected override fun createViewInstance(reactContext: ThemedReactContext): MLRNCallout {
        return MLRNCallout(reactContext);
    }
}
